flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/join/TemporalProcessTimeJoin.scala

```diff
@@ -60,32 +61,28 @@ class TemporalJoin(
     val rightStateDescriptor = new ValueStateDescriptor[Row]("right", rightType)
     rightState = getRuntimeContext.getState(rightStateDescriptor)
 
+    collector = new TimestampedCollector[CRow](output)
     cRowWrapper = new CRowWrappingCollector()
+    cRowWrapper.out = collector
   }
 
-  override def processElement1(
-      value: CRow,
-      ctx: CoProcessFunction[CRow, CRow, CRow]#Context,
-      out: Collector[CRow]): Unit = {
+  override def processElement1(element: StreamRecord[CRow]): Unit = {
 
     if (rightState.value() == null) {
       return
     }
 
-    cRowWrapper.out = out
-    cRowWrapper.setChange(value.change)
+    cRowWrapper.setChange(element.getValue.change)
+    collector.setTimestamp(element)
```

Comment:
Why call this function? The value of StreamRecord timestamp has already been erased.